src/gt4py/next/ffront/type_info.py

```diff
@@ -72,18 +72,120 @@ def return_type_fieldop(
     return ret_type
 
 
+@type_info.canonicalize_arguments.register
+def canonicalize_program_arguments(
+    program_type: ts_ffront.ProgramType,
+    args: tuple | list,
+    kwargs: dict,
+    *,
+    ignore_errors=False,
+    use_signature_ordering=False,
+) -> tuple[list, dict]:
+    return type_info.canonicalize_arguments(
+        program_type.definition,
+        args,
+        kwargs,
+        ignore_errors=ignore_errors,
+        use_signature_ordering=use_signature_ordering,
+    )
+
+
+@type_info.canonicalize_arguments.register
+def canonicalize_fieldop_arguments(
+    fieldop_type: ts_ffront.FieldOperatorType,
+    args: tuple | list,
+    kwargs: dict,
+    *,
+    ignore_errors=False,
+    use_signature_ordering=False,
+) -> tuple[list, dict]:
+    return type_info.canonicalize_arguments(
+        fieldop_type.definition,
+        args,
+        kwargs,
+        ignore_errors=ignore_errors,
+        use_signature_ordering=use_signature_ordering,
+    )
```

Comment:
You might consider shortening this to

```suggestion
@type_info.canonicalize_arguments.register(ts_ffront.FieldOperatorType)
@type_info.canonicalize_arguments.register(ts_ffront.ProgramType)
def canonicalize_program_arguments(
    gt_callable_type: ts_ffront.ProgramType | ts_ffront.FieldOperatorType,
    args: tuple | list,
    kwargs: dict,
    *,
    ignore_errors=False,
    use_signature_ordering=False,
) -> tuple[list, dict]:
    return type_info.canonicalize_arguments(
        gt_callable_type.definition,
        args,
        kwargs,
        ignore_errors=ignore_errors,
        use_signature_ordering=use_signature_ordering,
    )
```